### ISSUE TITLE
Static analysis warnings at build time: dead columns, unreachable columns, unused EDD fields

### ISSUE THREAD

**Issue by PaulSnow** [DTRules/DTRules#555]
## Why

Authors make mistakes that the runtime silently tolerates. Three patterns are common, detectable at `dtrules build` / XML-import time, and worth warning about:

1. Columns whose execution can never differ from another column's (no-op / redundant).
2. Columns whose condition pattern is obviously contradictory and can never match (unreachable).
3. EDD fields declared but never read or written by any condition or action in any decision table in the project.

All three are **warnings**, not errors. They print during `dtrules build` (or equivalent runtime import) and do not break CI. No new CLI command.

## Part 1 — Redundant / no-op columns

Per table. A column is flagged if it **cannot make an execution difference**:

- **No actions.** The column's action list is empty. (Unless the Y/N pattern has a test value that short-circuits other semantics — check table policy; for ALL tables an empty-action column contributes nothing.)
- **Action subsumption.** Another column has identical actions AND is always-also-matched whenever this one matches. Structural comparison of condition vectors is enough: if column `B`'s `(condition, required_value)` pairs are a superset (more permissive, fewer constraints) of column `A`'s, and `B`'s actions include all of `A`'s actions in order, then `A` is redundant.

Do structural detection only — compare DSL text and Y/N/- patterns verbatim. No SAT solver.

Warning form:

```
WARN <table_name>: column 3 is redundant (no actions) [no-op column]
WARN <table_name>: column 3 is redundant (subsumed by column 2) [no-op column]
```

## Part 2 — Unreachable columns

Per table. Flag obvious contradictions only:

- Two condition rows whose DSL text is an **identical negation** (e.g., `a > 5` vs `a <= 5`, or `a is equal to "X"` vs `a is not equal to "X"`), both required `Y` in the same column. Column is statically unreachable.
- Duplicate Y/N contradictions on the same condition (shouldn't happen — but defensive).

Don't try to solve general SAT. Miss cases involving arithmetic subsumption (`a > 5` + `a > 10` is redundant but not contradictory — that's Part 1, and we can skip it).

Warning form:

```
WARN <table_name>: column 4 can never match (conditions 2 and 5 are mutually exclusive) [unreachable column]
```

## Part 3 — Unused EDD fields

Cross **all tables in the project**. Walk every `_dt.xml` collecting referenced attributes (the identifiers used in `condition_dsl`, `action_dsl`, `context_dsl`, `initial_action_dsl`, and in test-data field names if that's part of the walk). Diff against the EDD's declared fields. Report fields declared but never referenced.

Carve-outs:

- Fields with attribute `access="r"` that are produced by a mapping (check `_map.xml`) are used — don't flag.
- `mapping*key` and other runtime-reserved fields — don't flag.
- Fields used only as assignment targets (written but never read) — still flag with different wording.

Warning form:

```
INFO unused EDD field: taxpayer.obscure_attribute (declared in TaxReturn_edd.xml, never referenced)
INFO write-only EDD field: job.temp_scratch (set but never read)
```

## Where this runs

During the XML-import step of `dtrules build`. Emit warnings to stderr. Keep `go test` output clean — don't fail any test.

## Implementation outline

- Per-table analyses (Parts 1 & 2): in `pkg/dtrules/decisiontable/` at build time, after the table structure is assembled.
- Project-wide analysis (Part 3): in `pkg/dtrules/sync/` or `cmd/dtrules/build.go`, after all DTs and the EDD are loaded. Use the already-parsed condition/action DSL text — no re-parsing needed.
- Collect warnings in a slice, print at the end of the build with a summary: `3 warnings (0 errors)`.

## Tests

- Fixtures under `pkg/dtrules/decisiontable/testdata/` (or similar):
  - A table with a no-action column → warning.
  - A table where column B subsumes column A → warning on A.
  - A table with a column containing a condition and its negation both required → unreachable warning.
  - A clean table → no warnings.
- EDD fixture with one declared-but-unused field → informational warning.
- Full TaxReturn project — sanity: capture the warnings produced, commit them as a golden file so drift is visible.

## Acceptance

- `dtrules build` on `sampleprojects/TaxReturn/` prints warnings for any existing no-ops / unreachables / unused fields. (Expected: several, given the scale of TaxReturn.)
- Warnings are additive only — exit code is unchanged unless pre-existing errors occur.
- Unit tests above all pass.

## Out of scope

- SAT-style semantic reasoning over condition expressions.
- Any attempt to auto-fix the flagged issues.
- Breaking CI on warnings.
